### DISCUSSION TITLE
Allow `if` conditions inside `services`

### DISCUSSION THREAD

**Discussion by 0ssigeno** [community/community#19680] [Actions]
Hi guys!
I’m writing at the moment Reusable Workflows for our organization. At the moment, different projects have different requirements at the services level. I would love to support that in the single Reusable Workflows, allowing to configure the services as well through its configuration. With the current expressive power, it is impossible to do it, because I can’t have a boolean condition on the creation of a service: for this use case, I would have to

- Always create all the services
- Have different job with all the different combination of services, sharing the same steps (with a composite action, but still)
- Use a step to write a docker-compose with the combination requested, and use that instead of the `services` offered by GA
- ???
In summary, what i would love to do in the Reusable Workflow is the following snippet
```
name: Reusable python tests workflow
on:
  workflow_call:
    inputs:
      use_postgres:
        description: Use the service container or not
        default: true
        type: boolean
        required: true
jobs:
    tests:
        name: Run python tests
        runs-on: ubuntu-latest
        timeout-minutes: 20
        services:
          postgres:
            if: ${{ inputs.use_postgres }}
            image: library/postgres:10.14-alpine
        steps:
              ...
```
